tokio/src/task/join_set.rs

```diff
@@ -53,6 +59,19 @@ pub struct JoinSet<T> {
     inner: IdleNotifiedSet<JoinHandle<T>>,
 }
 
+/// A variant of [`task::Builder`] that spawns tasks on a [`JoinSet`] rather
+/// than on the current default runtime.
+/// 
+/// [`task::Builder`]: crate::task::Builder
+#[cfg(all(tokio_unstable, feature = "tracing"))]
+#[cfg_attr(docsrs, doc(cfg(all(tokio_unstable, feature = "tracing"))))]
+#[must_use = "builders do nothing unless used to spawn a task"]
+#[derive(Debug)]
+pub struct Builder<'a, T> {
```

Comment:
This `#[derive(Debug)]` will put a `T: Debug` constraint on the impl, but it isn't actually necessary as `JoinSet` is `Debug` for non-Debug return types too.